### PR TITLE
[ref] use LocalizedText component to reduce the need for clien-side components

### LINKED ISSUE #202: tech discovery: see how we can manage i18n next.js

### PR DESCRIPTION
**Pull request by marco-souza**
## Description

To reduce the amount of components that we are rendering on the client-side `use client`, I created a `LocalizedText` component that will keep the state changes to the minimum, allowing us to remove a few client-rendered pages

Changes:
- **enh: create a component for translating text**
- **enh: update chakra version**
- **ref: localize text translation**

This closes #202

### CODE REVIEW

**Review by marco-souza — COMMENTED** [review]
With those changes, I was able to remove 4 `"use client";` 🎉

File: packages/components/Footer.tsx [hunk 1]
```diff
@@ -30,24 +28,29 @@ function PodCodarLogo() {
...
     <div className="flex flex-col items-center gap-2">
-      <p>{t("podcodar")}</p>
+      <p>
+        <LocalizedText translation={"footer.podcodar"} />
+      </p>
       <SocialIconLinks />
-      <p className="text-sm">{t("legal", { currentYear })}</p>
+      <p className="text-sm">
+        <LocalizedText translation="footer.legal" />
+        {currentYear}
```
Comment (Diegiwg) [inline packages/components/Footer.tsx:52]: While testing the preview version (https://webapp-git-marco-setup-localized-text-component-podcodar.vercel.app), the `currentYear` is not being rendered correctly. Could you please take a look and fix this?

image

File: packages/components/RoadmapSection.tsx [hunk 2]
```diff
@@ -34,7 +33,12 @@ export default function RoadmapSection() {
         my="2rem"
       >
         {cardList.map((cardProps) => (
-          <CardItem key={cardProps.title} color={cardProps.color} link={cardProps.link} title={t(cardProps.title)} />
+          <CardItem
```
Comment (Diegiwg) [inline packages/components/RoadmapSection.tsx:36]: Same issue of https://github.com/podcodar/webapp/pull/201/files#r1804012331

image

File: packages/components/MentoringSection.tsx [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
           </Heading>
 
           <Text color="gray.500" fontSize="3xl" textAlign="center" px="3rem">
-            {t("description")}
+            <LocalizedText translation="mentoring.description" />
           </Text>
         </GridItem>
         {mentoringList.map((mentoring) => (
-          <CardItem key={mentoring} title={t(`${mentoring}.title`)} description={t(`${mentoring}.description`)} />
+          <CardItem
```
Comment (Diegiwg) [inline packages/components/MentoringSection.tsx:24]: Same issue of https://github.com/podcodar/webapp/pull/201/files#r1804012331

image

**Review by marco-souza — COMMENTED** [review]
@Diegiwg thanks for catching that! 🙏🏼 
I just addressed your comments

**Review by Joel-leal — APPROVED** [review]

**Review by Diegiwg — APPROVED** [review]

**Pull request merged**